src/plugins/allowedMentions/AllowedMentions.tsx

```diff
@@ -0,0 +1,382 @@
+ */
+
+import "./AllowedMentions.css";
+
+import { Flex } from "@components/Flex";
+import { isNonNullish } from "@utils/guards";
+import { useForceUpdater } from "@utils/react";
+import { findByPropsLazy } from "@webpack";
+import { Clickable, Forms, GuildMemberStore, GuildStore, Menu, Popout as DiscordPopout, RelationshipStore, TextInput, useEffect, UserStore, useState } from "@webpack/common";
+import { Channel } from "discord-types/general";
+import { CSSProperties, ReactNode } from "react";
+
+export type AllowedMentionsParsables = "everyone" | "users" | "roles";
+
+export interface AllowedMentions {
+    parse: Set<AllowedMentionsParsables>;
+    users?: Set<string>;
+    roles?: Set<string>;
+    meta: {
+        hasEveryone: boolean;
+        userIds: Set<string>;
+        roleIds: Set<string>;
+        tooManyUsers: boolean;
+        tooManyRoles: boolean;
+    };
+}
+
+export interface AllowedMentionsProps {
+    mentions: AllowedMentions,
+    channel: Channel;
+    trailingSeparator?: boolean;
+}
+
+const replyClasses = findByPropsLazy("replyBar", "replyLabel", "separator");
+export const AllowedMentionsStore = {
+    store: new Map<string, AllowedMentions>(),
+    callbacks: new Map<string, (mentions: AllowedMentions) => void>,
+    get(channelId: string) {
+        return this.store.get(channelId);
+    },
+    set(channelId: string, mentions: AllowedMentions, dispatch: boolean) {
+        this.store.set(channelId, mentions);
+        dispatch && this.callbacks.get(channelId)?.(mentions);
+    },
+    delete(channelId: string) {
+        return this.store.delete(channelId);
+    },
+    clear() {
+        return this.store.clear();
+    },
+    subscribe(channelId: string, callback: (mentions: AllowedMentions) => void) {
+        return this.callbacks.set(channelId, callback);
+    },
+    unsubscribe(channelId: string) {
+        return this.callbacks.delete(channelId);
+    }
+};
+
+function getDisplayableUserNameParts(userId: string, guildId: string | null) {
+    // @ts-ignore discord-types doesn't have globalName
+    const { globalName, username } = UserStore.getUser(userId) ?? {};
+    const nickname = guildId ? GuildMemberStore.getNick(guildId, userId) : RelationshipStore.getNickname(userId);
+
+    return [nickname, globalName as string, username];
+}
+
+function getDisplayableUserName(userId: string, guildId: string | null) {
+    const [nickname, globalName, username] = getDisplayableUserNameParts(userId, guildId);
+
+    // Displayed name priority
+    // Guild/Friend Nickname > Global Name > Username
+
+    // User id if not cached in any stores
+    return nickname ?? globalName ?? username ?? userId;
+}
+
+function getDisplayableRoleName(roleId: string, guildId: string | null) {
+    // You *can* mention roles in DMs but is it really worth adding the UI for
+    // it in DMs
+    const role = guildId ? Object.values(GuildStore.getGuild(guildId).roles).find(r => r.id === roleId)?.name : undefined;
```

Comment:
You need to do this now.
```suggestion
    const role = guildId ? GuildStore.getRole(guildId, roleId)?.name : undefined;
```